tests/server/activity-event.js

```diff
@@ -0,0 +1,296 @@
+          originalUrl: '/',
+          query: {
+            client_id: 'mock client id', //eslint-disable-line camelcase
+            service: 'mock service',
+            zignore: 'ignore me'
+          }
+        });
+      },
+
+      teardown: function () {
+        process.stderr.write = write;
+      },
+
+      'process.stderr.write was called correctly': function () {
+        assert.equal(process.stderr.write.callCount, 1);
+
+        var eventData = JSON.parse(process.stderr.write.args[0][0].substr(14));
+        assert.lengthOf(Object.keys(eventData), 4);
+        assert.equal(eventData.service, 'mock service');
+      }
+    },
+
+    'call activityEvent without entrypoint, with entryPoint': {
+      setup: function () {
+        write = process.stderr.write;
+        process.stderr.write = sinon.spy();
+        return activityEvent('wibble', {}, {
+          headers: { 'user-agent': 'blee' },
+          originalUrl: '/',
+          query: {
+            entryPoint: 'mock entryPoint', //eslint-disable-line camelcase
+            ignore: 'ignore me'
+          }
+        });
+      },
+
+      teardown: function () {
+        process.stderr.write = write;
+      },
+
+      'process.stderr.write was called correctly': function () {
+        assert.equal(process.stderr.write.callCount, 1);
+
+        var eventData = JSON.parse(process.stderr.write.args[0][0].substr(14));
+        assert.lengthOf(Object.keys(eventData), 4);
+        assert.equal(eventData.event, 'wibble');
+        assert.equal(eventData.path, '/');
+        assert.equal(eventData.userAgent, 'blee');
+        assert.equal(eventData.entrypoint, 'mock entryPoint');
+      }
+    },
+
+    'call activityEvent with entrypoint, with entryPoint': {
+      setup: function () {
+        write = process.stderr.write;
+        process.stderr.write = sinon.spy();
+        return activityEvent('wibble', {}, {
+          headers: { 'user-agent': 'blee' },
+          originalUrl: '/',
+          query: {
+            entryPoint: 'mock entryPoint', //eslint-disable-line camelcase
+            entrypoint: 'mock entrypoint',
+            zignore: 'ignore me'
+          }
+        });
+      },
+
+      teardown: function () {
+        process.stderr.write = write;
+      },
+
+      'process.stderr.write was called correctly': function () {
+        assert.equal(process.stderr.write.callCount, 1);
+
+        var eventData = JSON.parse(process.stderr.write.args[0][0].substr(14));
+        assert.lengthOf(Object.keys(eventData), 4);
+        assert.equal(eventData.entrypoint, 'mock entrypoint');
+      }
+    },
+
+    'call activityEvent with 101-character query parameter': {
+      setup: function () {
+        write = process.stderr.write;
+        process.stderr.write = sinon.spy();
+        return activityEvent('wibble', {}, {
+          headers: { 'user-agent': 'blee' },
+          originalUrl: '/',
+          query: {
+            context: (new Array(102)).join('0')
```

Comment:
This just made my head explode. I've always used a loop to do this. So much more elegant.